api/v1/db/oracledb/game-dao.js

```diff
@@ -0,0 +1,65 @@
+const appRoot = require('app-root-path');
+const _ = require('lodash');
+
+const { serializeGames, serializeGame } = require('../../serializers/games-serializer');
+
+const conn = appRoot.require('api/v1/db/oracledb/connection');
+
+const sqlQuery = `
+    SELECT CN.CARD_NUMBER, CS.SUIT, G.GAME_ID, R.ROUND, G.MAXIMUM_BET,
+    G.MINIMUM_BET, G.BET_POOL
+    FROM GAMES G
+    INNER JOIN ROUNDS R ON G.ROUND_ID = R.ROUND_ID
+    LEFT OUTER JOIN TABLE_CARDS TC ON TC.GAME_ID = G.GAME_ID
+    LEFT OUTER JOIN CARDS C ON TC.CARD_ID = C.CARD_ID
+    LEFT OUTER JOIN CARD_NUMBERS CN ON C.CARD_NUMBER_ID = CN.CARD_NUMBER_ID
+    LEFT OUTER JOIN CARD_SUITS CS ON C.CARD_SUIT_ID = CS.SUIT_ID
+    `;
+/**
+ * @summary Return a list of games
+ * @function
+ * @param {Object} query query object that contains useful information to process.
+ * @returns {Promise<Object[]>} Promise object represents a list of games
+ */
+const getGames = async (query) => {
+  const connection = await conn.getConnection();
+  const round = query ? query.round : null;
+
+  try {
+    const sqlParams = {};
+    if (round) {
+      sqlParams.round = round;
+    }
+    const getGamesSqlQuery = `${sqlQuery} ${round ? 'WHERE R.ROUND = :round' : ''}`;
+    const rawGamesResponse = await connection.execute(getGamesSqlQuery, sqlParams);
+    const rawGames = rawGamesResponse.rows;
+    const serializedGames = serializeGames(rawGames, query);
+    return serializedGames;
+  } finally {
+    connection.close();
+  }
+};
+
+const getGameById = async (id) => {
+  const connection = await conn.getConnection();
+  try {
+    const sqlParams = [id];
+    const getGameByIdSqlQuery = `${sqlQuery} WHERE G.GAME_ID = :id`;
+    const rawGamesResponse = await connection.execute(getGameByIdSqlQuery, sqlParams);
+    const rawGames = rawGamesResponse.rows;
+    const groupedRawGames = _.groupBy(rawGames, 'GAME_ID');
+    if (_.isEmpty(groupedRawGames)) {
+      return undefined;
+    }
+    if (groupedRawGames.length > 1) {
```

Comment:
Could you take a look at what `_.groupBy()` returns again? This line isn't quite right. If you want to test this, try starting an interactive node session by calling `node`. Then, you can type
```
> const _ = require('lodash')
```
Try creating some simple test data and running `_.groupBy()` on it to see what is happening here